py3status/modules/google_calendar.py

```diff
@@ -0,0 +1,489 @@
+    {start_date} The start date for the event.
+    {start_time} The start time for the event.
+    {summary} The summary (i.e. title) for the event.
+    {time_to} The time until the event starts (or until it is over
+        if already in progress).
+
+Format_time_to and format_time_left placeholders:
+    {days} The number of days until the event.
+    {hours} The number of hours until the event.
+    {minutes} The number of minutes until the event.
+
+Color options:
+    color_separator: Color for separator.
+    color_event: Color for a single event.
+    color_time: Color for the time associated with each event.
+
+Requires:
+    1. Python library google-api-python-client (can be obtained with pip).
+    2. OAuth 2.0 credentials for the Google Calendar api.
+
+    Follow Step 1 of the guide here to obtain your OAuth 2.0 credentials:
+    https://developers.google.com/google-apps/calendar/quickstart/python
+
+    Download the client_secret.json file which contains your client ID and
+    client secret. In your i3status config file, set configuration parameter
+    client_secret to the path to your client_secret.json file.
+
+    The first time you run the module, a browser window will open asking you
+    to authorize access to your calendar. After authorization is complete,
+    an access/refresh token will be saved to the path configured in
+    auth_token, and i3status will be restarted. This restart will
+    occur only once after the first time you successfully authorize.
+
+Examples:
+```
+# add color gradients for events and dates/times
+google_calendar {
+    thresholds = {
+        'event': [(1, '#d0e6ff'), (2, '#bbdaff'), (3, '#99c7ff'), (4, '#86bcff'),
+            (5, '#62a9ff'), (6, '#8c8cff'), (7, '#7979ff')],
+        'time': [(1, '#ffcece'), (2, '#ffbfbf'), (3, '#ff9f9f'), (4, '#ff7f7f'),
+            (5, '#ff5f5f'), (6, '#ff3f3f'), (7, '#ff1f1f')]
+    }
+}
+```
+
+@author Igor Grebenkov
+@license BSD
+"""
+
+import httplib2
+import os
+import datetime
+import pytz
+
+from apiclient import discovery
+from oauth2client import client
+from oauth2client import clientsecrets
+from oauth2client import tools
+from oauth2client.file import Storage
+from httplib2 import ServerNotFoundError
+
+
+SCOPES = 'https://www.googleapis.com/auth/calendar.readonly'
+APPLICATION_NAME = 'py3status google_calendar module'
+
+
+class Py3status:
+    auth_token = '~/.config/py3status/google_calendar.auth_token'
+    button_open = 3
+    button_refresh = 2
+    button_toggle = 1
+    cache_timeout = 60
+    client_secret = '~/.config/py3status/google_calendar.client_secret'
+    force_lower_case = False
+    format = '{events}|'
+    format_date = '%a %d-%m'
+    format_event = '\?color=event {summary}[\?if=location  ({location})][\?if=time_to  {time_to}]'
+    format_event_full = '[\?color=event {summary}] ' +\
+        '[\?color=time ({start_time} - {end_time}, {start_date})]'
+    format_separator = '\?color=separator \| '
```

Comment:
The code might not be right if you have to place a space on right side... or hopefully it's just a typo.